SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -93,33 +110,78 @@ public async Task<IActionResult> UpdateUser(int id, [FromBody] User user)
                         await _context.SaveChangesAsync();
                         return Ok("User update successful");
                     }
-
                 }
             }
             else
             {
                 return new BadRequestObjectResult("user cannot be null");
             }
         }
+
         // POST: api/user/login
+        [AllowAnonymous]
         [HttpPost("login")]
-        public async Task<ActionResult> LoginPost([FromBody]User user)
+        public IActionResult CreateToken([FromBody]User user)
+        {
+            IActionResult response = Unauthorized();
+            var _user = Authenticate(user);
+            
+
+            if (_user != null)
+            {
+                var tokenString = BuildToken(_user);
+                response = Ok(new { token = tokenString });
+            }
+            return response;
+        }
+
+        private User Authenticate(User user)
         {
             if (user.Username == null || user.Password == null || user.Email == null)
             {
-                return new BadRequestObjectResult("No empty fields allowed");
+                new BadRequestObjectResult("No empty fields allowed");
```

Comment:
Shouldn't you throw something here if it's a bad request result?

Reply:
No. There's no need